amgut/handlers/add_sample.py

```diff
@@ -20,12 +21,24 @@ class AddSample(BaseHandler):
     @authenticated
     def post(self):
         # Required vars
-        barcode = self.get_argument('barcode')
-        sample_date = self.get_argument('sample_date')
-        sample_time = self.get_argument('sample_time')
-        notes = self.get_argument('notes')
-        participant_name = self.get_argument('participant_name')
-        sample_site = self.get_argument('sample_site', '')
+        participant_name = self.get_argument('participant_name',
+                                             'environmental')
+        form = self.build_form()
```

Comment:
These changes are so the form object is actually used to get and validate the logging information, most importantly so the date and time come back from the form as date and time objects. These are then fed into the newly retyped database columns so that everything stays properly typed and has less chance of being malformed, unusable data.